### PR TITLE
Modifying ServicesVisualStudio to dynamically load the extension mana…

### PR DESCRIPTION
**Pull request by tannergooding**
…ger based on the version of VS it is running on.

### CODE REVIEW

**Comment by tannergooding** [comment]
FYI. @jaredpar, @jasonmalinowski, @jmarolf, @amcasey, @Pilchie, @ManishJayaswal, @agocke 

This is for our SxS work and will allow VS to be built against VS2015 and should allow it to run against different versions of VS without a recompile.

The unit tests are not currently configured to do the same and still require VS2015 to run, but this will unblock our SxS work until that can be completed.

File: src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs [hunk 1]
```diff
@@ -28,8 +27,12 @@ internal class VisualStudioWorkspaceDiagnosticAnalyzerProviderService : IWorkspa
         [ImportingConstructor]
         public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorkspaceImpl workspace)
         {
+            // Microsoft.VisualStudio.ExtensionManager is non-versioned, so we need to dynamically load it, depending on the version of VS we are running on
+            var dteVersion = workspace.GetDte().Version;
+            var assembly = Assembly.Load($"Microsoft.VisualStudio.ExtensionManager, Version={dteVersion}, PublicKeyToken=b03f5f7f11d50a3a");
```
Comment (jaredpar) [inline src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs]: Use full version here.

File: src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs [hunk 2]
```diff
@@ -28,8 +27,12 @@ internal class VisualStudioWorkspaceDiagnosticAnalyzerProviderService : IWorkspa
         [ImportingConstructor]
         public VisualStudioWorkspaceDiagnosticAnalyzerProviderService(VisualStudioWorkspaceImpl workspace)
         {
+            // Microsoft.VisualStudio.ExtensionManager is non-versioned, so we need to dynamically load it, depending on the version of VS we are running on
```
Comment (jaredpar) [inline src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs]: Would add slight more detail heer.

**Comment by jaredpar** [comment]
:+1:

File: src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs (hunk 1 above)
Comment (tannergooding) [inline src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs]: Done.

File: src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs (hunk 2 above)
Comment (tannergooding) [inline src/VisualStudio/Core/Def/Implementation/Diagnostics/VisualStudioWorkspaceDiagnosticAnalyzerProviderService.cs]: Done.

**Comment by tannergooding** [comment]
Logged #5313 to track updating of the Unit Test assemblies.

**Comment by agocke** [comment]
:+1:

**Comment by Pilchie** [comment]
:+1:, apart from the part where Jenkins says it doesn't work :smile:

**Comment by tannergooding** [comment]
@Pilchie, @jaredpar, @agocke. Could you please review one more time. Using dynamic (which is simpler, cleaner, and easier) is not possible, because it cannot resolve members which are internal and exist in a separate assembly (InternalsVisibleTo is not used as part of the binding logic).

As such, I had to change this to use `object` and `Reflection`. I have confirmed that the Unit Tests pass locally and verified this is working SxS when patched into VS.

**Comment by Pilchie** [comment]
Still LGTM btw.

**Pull request merged**